Fixes from QA team 2018-03-12

realARCHaim opened the pull request:
- https://trello.com/c/v0l9S9C3
- https://trello.com/c/vKQ2hUU7
- https://trello.com/c/moLSPsTz
- https://trello.com/c/5jR4aJfL
- https://trello.com/c/Vd4KoyDb
- https://trello.com/c/IEM3pc4a
- https://trello.com/c/Sqy3sZZC

Commit:
Merge remote-tracking branch 'origin/development' into khaimovmr/fixes-from-qa-03-12

Commit:
Fixes from QA.

idegtiarov commented on askup/utils/general.py:
```diff
@@ -34,6 +35,9 @@
     ) as ranked
     {}
 '''
+PROFILE_RANK_USERS_COUNT = '''
```
Why multiple docstring? Usual string is seems to be fair enough.

realARCHaim replied:
You right. This construction is left from the first version of this feature. Now it's needed no more.

idegtiarov commented on askup/utils/general.py:
```diff
@@ -121,10 +125,13 @@ def get_user_profile_rank_list(rank_user_id, viewer_user_id):
 
     if user_is_present:
-        return result_row_datas
+        return result_row_datas, total_users
```
Function description and name are not relevant to the returned object any more, need to be updated

realARCHaim replied:
done

idegtiarov commented on askup/views.py:
```diff
@@ -191,13 +191,18 @@ def get_questions_queryset(self, context):
         return queryset
 
+    @staticmethod
+    def get_real_questions_queryset(qset_id):
```
why is this method used in frame of the class? I would make it the model's method or put into utils module. Because it is used in other view L752

realARCHaim replied:
Done. Moved into the utils general module.

idegtiarov commented on askup/views.py:
```diff
@@ -736,21 +747,48 @@ def answer_evaluate(request, answer_id, evaluation):
+    if question.id == previous_question.id:
+        return False
+
+    if question.vote_value != previous_question.vote_value:
```
@KhaimovMR could you please explaine the meaning of the following nex two statements? They are not clear enough, some comment strings would improve misunderstanding.

realARCHaim replied:
Rewriting the next question pick functionality for the cache one.

idegtiarov commented on requirements.txt:
```diff
@@ -3,3 +3,4 @@ django-crispy-forms==1.7.0
 django-debug-toolbar==1.9.1
 gunicorn==19.7.1
 psycopg2==2.7.3.2
+raven==6.6.0
```
I don't see in this PR any sentry configuration, does this change relates to this PR?

realARCHaim replied:
Removing this part.

idegtiarov reviewed (changes requested):
Please take a look at my comments and questions.

Commit:
New updates:

- Various PR 29 fixes
- Quiz remake with the cache
- Sentry configuration added

Commit:
Suggestions from PR 29:

- Sentry configuration fix
- Suggestions for quizzing

Commit:
More fixes.

idegtiarov commented on config/settings/base.py:
```diff
@@ -152,10 +152,21 @@
 SECRET_KEY = secure.SECRET_KEY
+
+if SENTRY:
+    import raven
```
why?

KhaimovMR commented:
https://trello.com/c/v0l9S9C3
https://trello.com/c/vKQ2hUU7
https://trello.com/c/moLSPsTz
https://trello.com/c/5jR4aJfL
https://trello.com/c/Vd4KoyDb
https://trello.com/c/IEM3pc4a
https://trello.com/c/Sqy3sZZC
